### PR TITLE
Add support for specifying location when creating KV stores

### PR DESCRIPTION
**Pull request by hrmsk66**
👋 This PR adds the ability to specify the location when creating a new KV store using Terraform.

Changes made:

- Update go-fastly to 9.3.0
- Add a new `location` field to the `fastly_kvstore` resource.
- Add a new acceptance test `TestAccFastlyKVStore_WithLocation_validate` and associated helper functions.
- Updated the document with the additional field.

### CODE REVIEW

File: fastly/resource_fastly_kvstore.go
Comment (Integralist) [inline fastly/resource_fastly_kvstore.go]: Let's validate the inputs using `ValidateDiagFunc` like this:
https://github.com/fastly/terraform-provider-fastly/blob/086f3dec206e4e7f8a1bd6aa030ae307901f0738/fastly/block_fastly_service_logging_s3.go#L39-L59

**EDIT:** I've just realised we didn't set up constants, so the values will need to be hard coded.

File: fastly/resource_fastly_kvstore.go
Comment (Integralist) [inline fastly/resource_fastly_kvstore.go]: Let's avoid setting `Default: ""` as I think Terraform will probably more correctly handle the null/nil case.

File: fastly/resource_fastly_kvstore.go
Comment (Integralist) [inline fastly/resource_fastly_kvstore.go]: Let's move this to a separate check like...

```go
input := &gofastly.CreateKVStoreInput{
	Name: d.Get("name").(string),
}

if v, ok := d.GetOk("location"); ok {
	input.Location = v.(string)
}
```

File: fastly/resource_fastly_kvstore_test.go
Comment (Integralist) [inline fastly/resource_fastly_kvstore_test.go:68]: Let's use the constants from go-fastly instead of hard coding the strings.]

**EDIT:** I've just realised we didn't set up constants 😬

**Review by Integralist — COMMENTED** [review]
Just a couple of minor comments, otherwise looks good 👍🏻

File: fastly/resource_fastly_kvstore_test.go
Comment (hrmsk66) [reply fastly/resource_fastly_kvstore_test.go:68]: @Integralist Does this approach look correct?
https://github.com/fastly/go-fastly/pull/521

Comment (Integralist) [reply fastly/resource_fastly_kvstore_test.go:68]: Yeah that's the general idea 👍🏻

Comment (hrmsk66) [reply fastly/resource_fastly_kvstore_test.go:68]: ~~Thank you for checking it. I have opened https://github.com/fastly/go-fastly/pull/521 🙏~~
~~Once the change is merged and released, I will update this PR.~~

**EDIT:** I have updated the PR based on your feedback. If you prefer to add constants to go-fastly first and then modify this PR accordingly, please feel free to hold off on merging this PR. There's no rush on my end.

File: fastly/resource_fastly_kvstore.go
Comment (Integralist) [inline fastly/resource_fastly_kvstore.go:32]:
```suggestion
			// TODO: Move values to constants inside of go-fastly.
			"location": {
```

**Review by Integralist — APPROVED** [review]
LGTM. I added a `TODO:` note into the code for future reference.

**Comment by hrmsk66** [comment]
Thank you 🙇‍♂️

**Pull request merged**
